Str.h

```diff
@@ -651,6 +652,49 @@ int     Str::appendf(const char* fmt, ...)
     return len;
 }
 
+void Str::replace(const char* find, const char* repl)
+{
+    STR_ASSERT(Owned == 1);
```

Comment:
Can take ownership after we know that replacements are needed (ideally without a double rewrite)